python/ray/rllib/dqn/replay_evaluator.py

```diff
@@ -63,38 +64,44 @@ def sample(self, no_replay=False):
             samples = [DQNEvaluator.sample(self)]
 
         for s in samples:
-            for obs, action, rew, new_obs, done in s:
-                self.replay_buffer.add(obs, action, rew, new_obs, done)
+            for row in s.rows():
+                self.replay_buffer.add(
+                    row["obs"], row["actions"], row["rewards"], row["new_obs"],
+                    row["dones"])
 
         if no_replay:
             return samples
 
         # Then return a batch sampled from the buffer
         if self.config["prioritized_replay"]:
```

Comment:
I hope later on all the ReplayBuffer casework can get pushed into the ReplayBuffer

Reply:
It's difficult. The prioritization requires extra computation that has to be part of the loss.